### PR TITLE
Getfeerate cmd options

### PR DESCRIPTION
**Pull request by SomberNight**
closes https://github.com/spesmilo/electrum/issues/2447#issuecomment-381360086

```
$ ./electrum getfeerate
26000
$ ./electrum getfeerate --fee_method "static" --fee_level 0.1
10000
$ ./electrum getfeerate --fee_method "static" --fee_level 0.4
50000
$ ./electrum getfeerate --fee_method "static" --fee_level 0.8
150000
$ ./electrum getfeerate --fee_method "eta" --fee_level 0.1
1081
$ ./electrum getfeerate --fee_method "eta" --fee_level 0.4
10705
$ ./electrum getfeerate --fee_method "eta" --fee_level 0.8
21190
$ ./electrum getfeerate --fee_method "mempool" --fee_level 0.1
1000
$ ./electrum getfeerate --fee_method "mempool" --fee_level 0.4
1000
$ ./electrum getfeerate --fee_method "mempool" --fee_level 0.8
20000
```

### CODE REVIEW

File: lib/commands.py [hunk 1]
```diff
@@ -657,10 +657,23 @@ def is_synchronized(self):
...
+        settings or supplied parameters.
+        """
+        if fee_method is None:
+            dyn, mempool = None, None
+        elif fee_method == 0:  # static
+            dyn, mempool = False, False
+        elif fee_method == 1:  # ETA
+            dyn, mempool = True, False
+        elif fee_method == 2:  # mempool
+            dyn, mempool = True, True
```
Comment (haarts) [inline lib/commands.py]: Why not dispense with the magic integers and just use the strings 'static'/'ETA'/'mempool'?

File: lib/commands.py [hunk 2]
```diff
@@ -657,10 +657,23 @@ def is_synchronized(self):
         return self.wallet.is_up_to_date()
 
     @command('n')
-    def getfeerate(self):
-        """Return current optimal fee rate per kilobyte, according
-        to config settings (static/dynamic)"""
-        return self.config.fee_per_kb()
+    def getfeerate(self, fee_method=None, fee_level=None):
```
Comment (haarts) [inline lib/commands.py]: Does it make sense to have `fee_level` default to `None`. Doing so complicates the code with several if statements. Also passing a default signals the argument is optional which it does seem to be.

File: lib/commands.py [hunk 3]
```diff
@@ -657,10 +657,23 @@ def is_synchronized(self):
...
+        """Return current suggested fee rate (in sat/kvByte), according to config
```
[the remark this answers is not in the document]
Comment (SomberNight) [reply lib/commands.py]: kilo-virtual-byte >.< looks weird, I know
1 vByte = 4 weight
1 kvByte = 4 kiloweight

File: lib/commands.py (hunk 2 above)
Comment (SomberNight) [reply lib/commands.py]: both arguments are optional. if they are not passed, we deduce their value from the config.
this also makes this change backwards compatible

File: lib/commands.py (hunk 1 above)
Comment (SomberNight) [reply lib/commands.py]: good point

**Pull request merged**